mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -56,9 +56,11 @@ def __init__(self, *,
             raise ValueError("Number of weights must match the number of optimization targets")
 
         self._space_adapter: Optional[BaseSpaceAdapter] = space_adapter
-        self._observations: List[Tuple[pd.DataFrame, pd.DataFrame, Optional[pd.DataFrame]]] = []
+        self._observations: List[Tuple[pd.DataFrame, pd.DataFrame, Optional[pd.DataFrame], Optional[pd.DataFrame]]] = []
```

Comment:
this gets a bit unwieldy. Should we have a named tuple for `_pending_observations`, e.g.,
```python
class PendingObservation(NamedTuple):
    """A named tuple representing a pending observation."""

    configurations: pd.DataFrame
    context: Optional[pd.DataFrame]
    meta: Optional[pd.DataFrame]
```
and do the same for `_observations`? (not sure we can inherit `NamedTuple`s - most likely, we can't)

Reply:
or, maybe, have `_observed_configs`, `_observed_scores`, `_observed_contexts` etc. and concatenate the dataframes instead of having a list of dataframes. I am pretty sure the schemas are the same from one `_register` call to the next